src/Search_Replace_Command.php

```diff
@@ -108,7 +116,50 @@ class Search_Replace_Command extends WP_CLI_Command {
 	 * : Produce report. Defaults to true.
 	 *
 	 * [--report-changed-only]
-	 * : Report changed fields only. Defaults to false.
+	 * : Report changed fields only. Defaults to false, unless logging, when it defaults to true.
+	 *
+	 * [--log[=<file>]]
+	 * : Log the items changed. If <file> is not supplied or is "-", will output to STDOUT.
+	 * Warning: causes a significant slow down, similar or worse to enabling --precise or --regex.
+	 *
+	 * [--log_context=<before_num,after_num>]
+	 * : Number of characters to display before and after the match. One number sets both before and after. Two comma-separated numbers set before and after respectively. Defaults to 40. Ignored if not logging.
+	 *
+	 * [--log_prefixes=<old_prefix,new_prefix>]
```

Comment:
Is this necessary as a command argument, or could it be an environment variable instead?

The command arguments available to `wp search-replace` have become a bit overwhelming. We may need to more actively address this in the future.

Reply:
It's easily changed to be an environment variable instead, so will do.

> The command arguments available to wp search-replace have become a bit overwhelming. 

Yes, the synopsis isn't easy to parse, especially with all those square brackets. Was thinking some auto-embolding (similar to most Unix man pages) would help but not sure:

**wp** **search-replace** <**old**> <**new**> [<**table**>...] [--**dry-run**] [--**network**] [--**all-tables-with-prefix]** [--**all-tables**] [--**export**[=<file>]] [--**export_insert_size**=<rows>] [--**skip-columns**=<columns>]
  [--**include-columns**=<columns>] [--**precise**] [--**recurse-objects**] [--**verbose**] [--**regex**] [--**regex-flags**=<regex-flags>] [--**regex-delimiter**=<regex-delimiter>] [--**format**=<format>] [--**report**]
  [--**report-changed-only**] [--**log**[=<file>]] [--**log_context**=<before_num,after_num>] [--**log_prefixes**=<old_prefix,new_prefix>] [--**log_colors**=<table_column_id_color,old_color,new_color>]

and similarly for the line-by-line options. I actually prototyped something like this a long time ago (when doing the word-wrapping) so will dig it up if you think it's worthwhile.